modules/solid_mechanics/doc/content/source/materials/CombinedPlasticityStressUpdate.md

```diff
@@ -0,0 +1,80 @@
+# Combined Plasticity Stress Update
+
+!syntax description /Materials/CombinedPlasticityStressUpdate
+
+## Description
+
+
+!include modules/solid_mechanics/common/supplementalRadialReturnStressUpdate.md
+
+In this model, the linear isotropic hardening function is given by $r = Q(1-exp(-bp))$, and kinematic
+hardening is governed by a backstress evolution equation of the form: $dX = Cdp - DXp$, where X is the backstress, and the two coefficients, C and D, represent the evolution of kinematic hardening.
+
+The effective plastic strain
+increment for combined hardening has the form:
+\begin{equation}
+ d \Delta p = \frac{\sigma^{trial}_{effective} - 3 G \Delta p - r - \sigma_{yield}}{3G + h}
+\end{equation}
+where $G$ is the isotropic shear modulus, and $\sigma^{trial}_{effective}$ is the scalar part
+of (von Mises trial stress - backstress).
+
+The non-linear equations for Isotropic and Kinematic Hardening are based on
+[!cite](besson2009non)
+pg. 82--84.
+
+This class calculates an effective trial stress, an effective scalar plastic strain increment, and
+the derivative of the scalar effective plastic strain increment; these values are passed to the
+[BackstressRadialReturnStressUpdate.md] to compute the radial return stress
+increment.  This plasticity class also computes the plastic strain as a stateful material
+property.
+
+This class is based on the implicit integration algorithm in [!cite](dunne2005introduction)
+pg. 146--149.
+
+The `ADCombinedPlasticityStressUpdate` version of this class uses forward mode automatic
+differentiation to provide all necessary material property derivatives to
+assemble a perfect Jacobian (this replaces the approximated tangent operator).
+
+This model can be reduced to a purely linear isotropic model, if the kinematic hardening modulus is set to zero, as the backstress in the effective trial stress will remain zero and will not be updated. Conversely, if the isotropic hardening constant is set to zero, the hardening model will behave purely as linear kinematic.
+
+To model the nonlinear hardening behavior of a material, the constants Q (saturation hardening) and b (rate of hardening) should be set to nonzero values for nonlinear isotropic hardening. Additionally, to capture nonlinear kinematic hardening behavior, the material constant gamma should be nonzero. For a combined hardening model, all constants (Q, b, and gamma) should be set to nonzero values.
```

Comment:
You don't have to set all of those constants to nonzero values for combined hardening -- you could set a couple of them and get combined linear hardening.

Reply:
Q and b are nonlinear isotropic coefficients. 
kinematic saturation gamma is a nonlinear kinematic coefficient.
To get combined **nonlinear** hardening you set all the values to nonzero. 
For **linear** combined hardening, you don't need any of the three parameters.